src/zmalloc.c

```diff
@@ -87,16 +94,39 @@ static size_t used_memory = 0;
 pthread_mutex_t used_memory_mutex = PTHREAD_MUTEX_INITIALIZER;
 
 #ifdef USE_MEMKIND
-static void* (*pmem_malloc)(size_t size) = NULL;
-static void (*pmem_free)(void* ptr) = NULL;
-static void* (*pmem_realloc)(void* ptr,size_t size) = NULL;
-void zmalloc_init_pmem_functions(void* (*_pmem_malloc)(size_t),
-                            void (*_pmem_free)(void*),
-                            void* (*_pmem_realloc)(void*,size_t)){
-
-    pmem_malloc = _pmem_malloc;
-    pmem_free = _pmem_free;
-    pmem_realloc = _pmem_realloc;
+static struct memkind *pmem_kind;
+
+//return 0 if pmem else return 1
+int zmalloc_check_if_is_pmem (void * ptr)
```

Comment:
Please rename function to something like:
"zmalloc_get_location" where return type is defined as (example):
#DEFINE DRAM_LOCATION 0
#DEFINE PMEM_LOCATION 1
#DEFINE WRONG_LOCATION 2

Reply:
Done.